src/components/Campaign.tsx

```diff
@@ -17,18 +16,21 @@ export const editorTopBarTestId = "editor-top-bar-message";
 
 export const Campaign = () => {
   const { idCampaign } = useParams() as Readonly<{ idCampaign: string }>;
-  const { setContent, unsetContent, getContent } = useSingletonEditor();
-
   const campaignContentQuery = useGetCampaignContent(idCampaign);
   const campaignContentMutation = useUpdateCampaignContent();
+  const { getContent, save } = useSingletonEditor(
+    {
+      initialContent: campaignContentQuery.data,
+      onSave: async () => {
+        const content = await getContent();
+        campaignContentMutation.mutate({ idCampaign, content });
+      },
+    },
+    [campaignContentQuery.data]
+  );
```

Comment:
No estoy seguro, pero por ahí que el `onSave` reciba el contendocomo parámetro podría simplifcar el código a futuro?

```suggestion
  const { getContent, save } = useSingletonEditor(
    {
      initialContent: campaignContentQuery.data,
      onSave: async (content) => {
        campaignContentMutation.mutate({ idCampaign, content });
      },
    },
    [campaignContentQuery.data]
  );
```

Reply:
Si es trivial, creo que podría ir en este PR, sino, lo dejaría para otro.

Reply:
Hice una refactorización sobre el método **_onSave_** que pasábamos desde el **_SingletonEditorProvider_**, esto me ayudó a encapsular el **_getContent_** como comentamos antes en este hilo.

Reply:
Buenisimo! marco la conversación como resuelta.